src/App.jsx

```diff
@@ -14,33 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'Sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sort, setSort] = useState({ type: '', reversed: false });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  // function for sort by length and alphabet
+  const handleSort = type => {
+    const sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (type === SORT_BY_ALPHABET) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (type === SORT_BY_LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (sort.reversed) {
+      sortedGoods.reverse();
+    }
+
+    setSort({ type, reversed: sort.reversed });
```

Comment:
When a new sort type is selected, the `reversed` state should be reset to `false` to avoid unexpected behavior. Consider updating the `setSort` call to `setSort({ type, reversed: false });` to ensure the list is not reversed by default when a new sort type is applied.